types.go

```diff
@@ -74,13 +75,16 @@ type Level3Data struct {
 type AuthorizationRequest struct {
 	Amount                     Amount
 	CreditCard                 *CreditCard
-	BillingAddress             *BillingAddress
+	BillingAddress             *Address
 	Level3Data                 *Level3Data
 	ClientTransactionReference *string // Custom transaction reference metadata that will be associated with this request
 	Channel                    string  // for Psps that track the sales channel
 	Cryptogram                 string  // for Network Tokenization methods
 	ECI                        string  // E-Commerce Indicator (can be used for Network Tokenization as well)
 	MerchantOrderReference     string  // Similar to ClientTransactionReference but specifically if we want to store the shopping cart order id
+	ShippingAddress            *Address
+	ShopperIP                  *string
+	ShopperEmail               *string
```

Comment:
Should we use email from shipping or billing address instead of creating a new field for this?

Reply:
I can see billing address, but probably not shipping.

With limited context I have so far maybe we could pull it from billing address. But I think its best to let whoever is using the library choose.

Reply:
Since the ShopperEmail field we are passing is being used as a risk signal / in a velocity rule by Adyen, I think we should pull it from billing address since billing address ties directly into payments. Also, having three different places to pass a shopper email address is going to lead to confusion.